runner/execer/os/execer.go

```diff
@@ -30,20 +30,34 @@ type WriterDelegater interface {
 // Implements runner/execer.Execer
 type execer struct {
 	// Best effort monitoring of command to kill it if resident memory usage exceeds this cap
-	memCap scootexecer.Memory
-	stat   stats.StatsReceiver
-	pw     ProcessWatcher
+	memCap            scootexecer.Memory
+	getMemUtilization func(int) (scootexecer.Memory, error)
+	stat              stats.StatsReceiver
+	pw                ProcessWatcher
 }
 
-// NewBoundedExecer returns an execer with a ProcGetter and, if non-zero values are provided, a memCap and a StatsReceiver
-func NewBoundedExecer(memCap scootexecer.Memory, stat stats.StatsReceiver) *execer {
+// NewBoundedExecer returns an execer with a ProcGetter and, if non-zero values are provided, a memCap, overriding memory utilization function, and a StatsReceiver
+func NewBoundedExecer(memCap scootexecer.Memory, getMemUtilization func() (int64, error), stat stats.StatsReceiver) *execer {
 	oe := &execer{pw: NewProcWatcher()}
 	if memCap != 0 {
 		oe.memCap = memCap
 	}
 	if stat != nil {
 		oe.stat = stat
 	}
+	// if not nil, use the provided function to get memory utilization,
+	// otherwise get the memory usage of the current process and its subprocesses
+	if getMemUtilization != nil {
+		oe.getMemUtilization = func(int) (scootexecer.Memory, error) {
+			mem, err := getMemUtilization()
+			if err != nil {
+				return 0, err
```

Comment:
nit: would -1 make more sense here

Reply:
scootexecer.Memory is of type uint64, hence made it 0 as we receive -1 if there's an error.